informer/src/informer_service/rate_limiter.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::time::Duration;
+use tokio::time;
+
+pub struct RateLimiter {
+    ticket: u64,
+    pub current_ticket: u64,
+}
+
+impl RateLimiter {
+    pub fn new(ticket: u64) -> Self {
+        Self {
+            ticket,
+            current_ticket: 0,
+        }
+    }
+
+    pub async fn acquire_ticket(&mut self) {
+        self.current_ticket += 1;
+        if self.current_ticket == self.ticket {
+            self.current_ticket = 0;
+            time::delay_for(Duration::from_millis(1000)).await;
```

Comment:
Current implementation sleeps 1 second every N tickets.
How about sleeping if tickets acquired more than maximum in a second?

Reply:
For example

```rs
self.current_ticket += 1;
let remain
let elapsed = self.last_reset.elapsed();
if elapsed < Duration::from_secs(1) && self.current_ticket == self.maximum_ticket {
  time::delay_for(Duration::from_secs(1) - elapsed).await;
}
```